internal/n/crit.go

```diff
@@ -0,0 +1,94 @@
+package n
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/capatazlib/go-capataz/internal/c"
+	"github.com/capatazlib/go-capataz/internal/s"
+)
+
+// EventCriteria is an utility that allows us to specify a matching criteria to
+// a specific supervision event
+type EventCriteria func(s.Event) bool
+
+// EAnd joins a slice of EventCriteria with an and statement
+func EAnd(crits ...EventCriteria) EventCriteria {
+	return func(ev s.Event) bool {
+		result := true
+		for _, crit := range crits {
+			result = result && crit(ev)
+			if !result {
+				return result
+			}
+		}
+		return result
+	}
+}
+
+// EOr joins a slice of EventCriteria with an or statement
```

Comment:
You should note that an empty criteria is defined as _false_. `empty || true -> true`, `empty || false -> false`